src/modules/commander/HealthAndArmingChecks/Common.hpp

```diff
@@ -74,6 +74,12 @@ static_assert(sizeof(navigation_mode_group_t) == sizeof(NavModes), "type mismatc
 static_assert(vehicle_status_s::NAVIGATION_STATE_MAX <= CHAR_BIT *sizeof(navigation_mode_group_t),
 	      "type too small, use next larger type");
 
+// Type to pass twe mode groups in one struct to have the same number of arguments
+struct NavModesMessageFail {
+	NavModes message_modes; ///< modes in which there's user messageing but arming is allowed
+	NavModes fail_modes; ///< modes in which arming checks fail (subset of message_modes)
```

Comment:
It being a subset is not enforced, is it? Maybe better to make that clear here, e.g. "should be subset"

Reply:
It is enforced as in messages are sent for `message_modes | fail_modes` which adds messaging for all modes in which arming is blocked but there would not be any message. I change the version slightly.